picturae_importer.py

```diff
@@ -919,33 +926,37 @@ def upload_records(self):
         """
         # the order of operations matters, if you change the order certain variables may overwrite
 
-        self.record_full = self.record_full[self.record_full['CatalogNumber'].isin(self.barcode_list)]
-
         self.record_full = self.record_full.drop_duplicates(subset=['CatalogNumber'])
 
         for row in self.record_full.itertuples(index=False):
-            self.populate_fields(row)
+            if row.CatalogNumber in self.barcode_list:
+                self.populate_fields(row)
+
```

Comment:
now will change the barcode_present = True when each row is processed to prevent the csv from accidently being uploaded twice( in case it fails critically enough to skip creating a picbatch record, making it a pain to undo)